coref/const.py

```diff
@@ -134,13 +134,17 @@ def create_simulation_pseudodoc(self) -> "Doc":
             ),
         )
 
-    def subwords_2_words(self, subwords: Iterable[int]) -> set[int]:
+    def subwords_2_words_w_payload(
+        self, subwords: Iterable[int], payload: Iterable[float]
+    ) -> list[Tuple[int, float]]:
         words_2_subwords_map = {
             subtoken: i
             for i, (start, end) in enumerate(self.word2subword)
             for subtoken in range(start, end)
         }
-        return set(words_2_subwords_map[subword] for subword in subwords)
+        res = [(words_2_subwords_map[s], p) for s, p in zip(subwords, payload)]
+
+        return [(k, np.mean(group)) for k, group in res]
 
```

Comment:
Am I reading it wrong? But I see that the mean value of a single float is computed at each iteration?

Reply:
Also, 
<img width="1060" alt="Screenshot 2023-03-31 at 1 36 19" src="https://user-images.githubusercontent.com/26472774/228987779-c5e5bb91-ab85-4b16-96a1-6aa0ae662109.png">

Reply:
there is a grouping step missing... Good catch